src/pages/_app.tsx

```diff
@@ -1,11 +1,27 @@
-import { AppProps } from "next/app";
-import { ApolloClient, InMemoryCache } from "@apollo/client";
+import { ApolloClient, concat, HttpLink, InMemoryCache } from "@apollo/client";
+import { setContext } from "@apollo/client/link/context";
 import { ApolloProvider } from "@apollo/client/react";
-import { AuthProvider } from "../../utils/firebase/auth";
+import { AppProps } from "next/app";
 import "tailwindcss/tailwind.css";
+import { AuthProvider } from "../../utils/firebase/auth";
+import firebase from "../../utils/firebase/firebase";
+
+const httpLink = new HttpLink({ uri: process.env.NEXT_PUBLIC_API_URL });
+
+const authLink = setContext(async (_, { headers }) => {
+  const token = await firebase
+    .auth()
+    .currentUser?.getIdToken(/* forceRefresh */ false);
+  return {
```

Comment:
Is "headers" going to be the only field passed through the link? I would unpack the context as {headers, ...context} and pass other context forward as well.